src/quo/components/share/share_qr_code/view.cljs

```diff
@@ -121,91 +125,123 @@
     :as   props}]
-  [:<>
-   [rn/view {:style style/gradient-bg}
-    [gradient-cover/view {:customization-color customization-color :height 463}]]
-   [rn/view {:style style/content-container}
-    [rn/view
-     {:style style/share-qr-container}
-     [rn/view
-      {:style style/share-qr-inner-container}
-      [account-avatar/view
-       {:customization-color customization-color
-        :emoji               emoji
-        :size                32}]
-      [text/text
-       {:size   :heading-2
-        :weight :semi-bold
-        :style  {:margin-left 8}} full-name]]
-     [share-button {:on-press on-share-press}]]
-    (when (#{:wallet-legacy :wallet-multichain} share-qr-type)
-      [header props])
-    [quo.theme/provider {:theme :light}
-     [qr-code/view
-      {:qr-image-uri        qr-image-uri
-       :size                (style/qr-code-size component-width)
-       :avatar              (if (= share-qr-type :profile)
-                              :profile
-                              :wallet-account)
-       :customization-color customization-color
-       :full-name           full-name
-       :profile-picture     profile-picture
-       :emoji               emoji}]]
-    [rn/view {:style style/bottom-container}
-     (case share-qr-type
-       :profile           [profile-bottom props]
-       :wallet-legacy     [wallet-legacy-bottom props]
-       :wallet-multichain [wallet-multichain-bottom props]
-       nil)]]])
+  [rn/view {:style style/content-container}
+   [rn/view {:style style/share-qr-container}
+    [rn/view {:style style/share-qr-inner-container}
+     [header-icon props]
+     [text/text
+      {:size   :heading-2
+       :weight :semi-bold
+       :style  {:margin-left 8}} full-name]
+     (when (= share-qr-type :watched-address)
+       [icons/icon
+        :i/reveal
+        {:color           colors/white-opa-40
+         :container-style style/watched-account-icon}])]
+    [share-button {:on-press on-share-press}]]
+   (when (not= share-qr-type :profile)
+     [header props])
+   [quo.theme/provider {:theme :light}
+    [qr-code/view
+     {:qr-image-uri        qr-image-uri
+      :size                (style/qr-code-size component-width)
+      :avatar              (case share-qr-type
+                             :profile                   :profile
+                             (:watched-address :wallet) :wallet-account
+                             :saved-address             :saved-address
+                             nil)
+      :customization-color customization-color
+      :full-name           full-name
+      :profile-picture     profile-picture
+      :emoji               emoji}]]
+   [rn/view {:style style/bottom-container}
+    (if (= share-qr-type :profile)
+      [profile-bottom props]
+      (case address
+        :legacy     [wallet-legacy-bottom props]
+        :multichain [wallet-multichain-bottom props]
+        nil))]])
 
 (defn- view-internal
   "Receives the following properties:
```

Comment:
can we remove this doc string and instead use Malli?
@ilmotta, @clauxx and @ajayesivan can help if there's any questions about it. Should be some simple examples to follow too :)

Reply:
👍 a malli schema would be great here. The props here are unusually complex, with conditional properties based on the :type prop. Can be confusing if there will be changes in the future and forget to update the docstring. With a schema, it will break the tests in that case.

Reply:
@J-Son89. @clauxx implemented it 👌